smartapps/tonesto7/nest-manager.src/nest-manager.groovy

```diff
@@ -561,6 +574,9 @@ def updateChildData() {
             	return true
         	} 
             
+        	else if(devId == "NestPresenceDevice") {
```

Comment:
There was a race condition that getchilddevices returns children as still present shortly after they have been deleted.

This catches this if you just deleted the presence sensor.  Otherwise you get an exception.

It seems deletechildren takes a few seconds, and the code runs faster than that.

This is a problem when you add devices and delete the presence sensor in the same update operation.

It actually happens for other devices also - the code just tries to send them the update.

On Mar 22, 2016, at 11:47 PM, Tonesto7 <notifications@github.com<mailto:notifications@github.com>> wrote:


In smartapps/tonesto7/nest-manager.src/nest-manager.groovy<https://github.com/tonesto7/nest-manager/pull/33#discussion_r57106409>:

> @@ -561,6 +574,9 @@ def updateChildData() {
>               return true
>               }
>
> +             else if(devId == "NestPresenceDevice") {


Just curious what's the need for this? Versus the else if above it?

—
You are receiving this because you authored the thread.
Reply to this email directly or view it on GitHub<https://github.com/tonesto7/nest-manager/pull/33/files/42c88c6d13dc0dc7c519d71c49f5ea64862e4efc#r57106409>